### PR TITLE
Update DROP TABLE in Iceberg document

### PR DESCRIPTION
**Pull request by agrawalreetika**
## Description
Update DROP TABLE in Iceberg document

## Motivation and Context
Refactor existing Iceberg document about DROP TABLE

## Impact
After Change -
image

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -482,11 +482,13 @@ Alter table operations are supported in the connector::
 DROP TABLE
 ^^^^^^^^^^^
 
-Drop the table ``page_views``. This only drops the metadata
-& data for the table::
+Drop the table ``page_views`` ::
 
     DROP TABLE iceberg.web.page_views
 
+* Dropping an Iceberg table with ``Hive Metastore`` & ``Glue`` Catalogs only removes metadata from metastore.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:489]:
```suggestion
* Dropping an Iceberg table with Hive Metastore & Glue catalogs only removes metadata from metastore.
```

Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:489]:
```suggestion
* Dropping an Iceberg table with Hive Metastore & Glue catalogs only removes metadata from metastore.
```

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -482,11 +482,13 @@ Alter table operations are supported in the connector::
...
+* Dropping an Iceberg table with ``Hadoop`` & ``Nessie`` Catalogs removes all the data & metadata in the table.
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:490]:
```suggestion
* Dropping an Iceberg table with Hadoop and Nessie catalogs removes all the data and metadata in the table.
```

**Review by steveburnett — CHANGES_REQUESTED** [review]
Nice! I made some formatting suggestions, I wanted to explain why I made these suggestions here.
 
I don't think we need to mark Hive Metastore, Glue, Hadoop, and Nessie here as code because they're just nouns here. 

I changed "&" to "and" because Latin abbreviations such as &, i.e., e.g, and such should not be in technical documentation. For one example, see the [GitLab Recommended word list](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#) entry for &.

**Comment by agrawalreetika** [comment]
Thank you for details @steveburnett 
I have made the suggested changes. Please review.

**Review by steveburnett — APPROVED** [review]
LGTM! (docs) Changes verified in local build. Thanks @agrawalreetika!

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**